### PR TITLE
Add support for zone-level routes to Access resources

### PR DESCRIPTION
**Pull request by Justin-Holmes**
Requires a new cloudflare-go version with https://github.com/cloudflare/cloudflare-go/pull/538 before this can be merged.

Closes https://github.com/cloudflare/terraform-provider-cloudflare/issues/812.

### CODE REVIEW

**Comment by Justin-Holmes** [comment]
@jacobbednarz would love your opinion on this approach before I implement this for the remaining resources!

File: cloudflare/resource_cloudflare_access_application.go [hunk 1]
```diff
@@ -28,10 +28,9 @@ func resourceCloudflareAccessApplication() *schema.Resource {
 				Computed: true,
 			},
 			"zone_id": {
-				Deprecated: "This field will be removed in version 3 and replaced with the account_id field.",
-				Type:       schema.TypeString,
-				Optional:   true,
-				Computed:   true,
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
```
Comment (jacobbednarz) [inline cloudflare/resource_cloudflare_access_application.go:33]: Should we add some schema level validation here via `ConflictsWith: []string{"other_attr"}` to make sure that we only ever have `zone_id` or `account_id`?

**Review by jacobbednarz — COMMENTED** [review]
I'm on board with this approach 👍 My initial concerns were that this is going to get unwieldy with these conditionals in each CRUD method however I can't think a better way of getting it without sacrificing readability and introducing some ✨ magic methods ✨ but it will make life pretty hard to debug when the time comes so let's give this a whirl.

**Review by patryk — CHANGES_REQUESTED** [review]
Need documentation changes and remove go.mod/sum from the PR, then we're good to merge.

**Comment by jacobbednarz** [comment]
A thought for the documentation - It would be great if we could explicitly call out which uses cases fit `account_id` and `zone_id` so that we can have that as a reference point when debugging or digging into any issues that pop up.

**Comment by Justin-Holmes** [comment]
@jacobbednarz Sure thing! Do you have a particular place in the documentation you'd like to me to add the notes?

**Comment by jacobbednarz** [comment]
Nothing specific in mind so happy to defer to your best judgement on what explains it best. I'm just after an easy reference/guide for determining which one should be used. Some examples that come to mind are things like:

- Full HCL examples;
- Note in the attributes reference section; 
- Callout on the documentation (like https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/certificate_pack). Available callouts are documented at https://github.com/hashicorp/terraform-website#callouts

**Review by patryk — APPROVED** [review]

**Pull request merged**